test/org/sagebionetworks/bridge/services/StudyServiceMockTest.java

```diff
@@ -1459,6 +1461,47 @@ public void testAllFourTemplatesAreSanitized() {
         assertHtmlTemplateSanitized( study.getVerifyEmailTemplate() );
         assertHtmlTemplateSanitized( study.getAccountExistsTemplate() );
     }
+    
+    @Test
+    public void updateStudyCorrectlyDetectsEmailChangesInvolvingNulls() {
+        // consent email still correctly detected
+        String originalEmail = TestUtils.getValidStudy(StudyServiceMockTest.class).getConsentNotificationEmail();
+        String newEmail = "changed@changed.com";
+        
+        setupConsentEmailChangeTest(null, null);
+        setupConsentEmailChangeTest(originalEmail, originalEmail);
+        setupConsentEmailChangeTest(null, newEmail);
+        setupConsentEmailChangeTest(originalEmail, null);
+        setupConsentEmailChangeTest(originalEmail, newEmail);
+    }
+    
+    private void setupConsentEmailChangeTest(String originalEmail, String newEmail) {
+        reset(sendMailService);
+        Study original = TestUtils.getValidStudy(StudyServiceMockTest.class);
+        original.setConsentNotificationEmail(originalEmail);
+        when(studyDao.getStudy(any())).thenReturn(original);
+        
+        Study update = TestUtils.getValidStudy(StudyServiceMockTest.class);
+        update.setConsentNotificationEmail(newEmail);
+        // just assume this is true for the test so defaults aren't set
+        update.setConsentNotificationEmailVerified(true);
+        
+        service.updateStudy(update, true);
+        
+        boolean hasChanged = !Objects.equals(original.getConsentNotificationEmail(),
+                update.getConsentNotificationEmail());
```

Comment:
The test and the code both have the same code. If there's a bug in the code, the same bug is present in the test. Instead of trying to calculate the hasChanged, we should pass in the expected result and check against that instead.

Reply:
Changed to specify the expected result